sbom/cyclonedx.go

```diff
@@ -168,30 +169,59 @@ func (ccx *CycloneDX) convertCycloneDxToSbom(bom *cyclonedx.BOM) (*Sbom, error)
 		return nil, errors.New("not a valid cyclone dx BOM")
 	}
 
+	rootComponent := bom.Metadata.Component
 	sbom := &Sbom{
 		Asset: &Asset{
-			Name: bom.Metadata.Component.Name + ":" + bom.Metadata.Component.Version,
+			Name: rootComponent.Name,
+			Platform: &Platform{
+				Version: rootComponent.Version,
+				Title:   rootComponent.Description,
+			},
 		},
 		Packages: make([]*Package, 0),
 	}
 
-	if bom.Metadata.Tools != nil && bom.Metadata.Tools.Components != nil {
-		// last one wins :-) - we only support one tool
-		for _, component := range *bom.Metadata.Tools.Components {
-			sbom.Generator = &Generator{
-				Name:    component.Name,
-				Version: component.Version,
-				Vendor:  component.Author,
+	switch rootComponent.Type {
+	case cyclonedx.ComponentTypeOS:
+		hostnameId := "//platformid.api.mondoo.app/hostname/" + rootComponent.Name
+		sbom.Asset.PlatformIds = append(sbom.Asset.PlatformIds, hostnameId)
+	case cyclonedx.ComponentTypeContainer:
+		// we need to figure out where to get the container ID from properly. For now, we use the BOMRef
+		bomRefId := "//platformid.api.mondoo.app/runtime/docker/images/" + rootComponent.BOMRef
+		sbom.Asset.PlatformIds = append(sbom.Asset.PlatformIds, bomRefId)
+	}
+
+	if bom.Metadata.Tools != nil {
+		if bom.Metadata.Tools.Components != nil {
+			// last one wins :-) - we only support one tool
+			for _, component := range *bom.Metadata.Tools.Components {
+				sbom.Generator = &Generator{
+					Name:    component.Name,
+					Version: component.Version,
+					Vendor:  component.Author,
+				}
+			}
+		}
+
+		// if we have no generator info, fallback to trying tools. these are deprecated
+		// but might still be present
+		if sbom.Generator == nil && bom.Metadata.Tools.Tools != nil {
```

Comment:
this is purely fallback in case the tools are present and we couldnt get a generator